openshift/system/config/cache_store.yml

```diff
@@ -1,3 +1,4 @@
 production:
   - :mem_cache_store
   - <%= ENV['MEMCACHE_SERVERS'].presence || 'system-memcache:11211' %>
+  - :digest_class: !ruby/class Digest::SHA256
```

Comment:
Is this required? Can we set it as some default value? Just wondering, because if this is required, we need to update SaaS configs also

Reply:
I think so. The digest algorithm was being configured through a monkey patch I removed, so it needs to be configured somehow. Maybe there are other options like setting it from an initializer or applicaton.rb, but I think config files are the best option.

Reply:
What monkey patch was this? Also, I can't find any reference to this setting in Rails documentation... 
But in any case, if this is the default option that we are not planning to change across environments, I'd really prefer to have is somewhere here: https://github.com/3scale/porta/blob/dbdf5b570a960982e841b9c0b03f60e29e613bb6/config/application.rb#L189

Reply:
I think @jlledom doesn't mean monkey patch but this:

https://github.com/3scale/porta/pull/2971/files#diff-c1fd91cb1911a0512578b99f657554526f3e1421decdb9e908712beab57e10f9R51-R52

Reply:
If that's the case - we still have it here: https://github.com/3scale/porta/pull/3456/files#diff-c1fd91cb1911a0512578b99f657554526f3e1421decdb9e908712beab57e10f9R71-R73

I guess that applies for all ActiveSupport, cache included? (although, no idea)

Reply:
Sorry, I meant this: https://github.com/3scale/porta/commit/442b606278e8c632dfdee12352e298e4ebb89361#diff-ab512545d0eb817a27d443feddbdcf22962c873d891a20c8cc7fe63e5b0e9656L13

The options given to the config file are read from: 

https://github.com/3scale/porta/blob/dbdf5b570a960982e841b9c0b03f60e29e613bb6/config/application.rb#L189

so we end up calling this:

`config.cache_store = [:mem_cache_store, ["localhost:11211"], {:digest_class=>Digest::SHA256}]`

@mayorova If you really don't like this, I would prefer to set the digest in an initializer (I guess it's possible). But I would avoid adding more complexity to the cache section in `application.rb`

@akostadinov regarding this: https://github.com/3scale/porta/pull/2971/files#diff-c1fd91cb1911a0512578b99f657554526f3e1421decdb9e908712beab57e10f9R51-R52

`ActiveSupport::Digest.hash_digest_class = OpenSSL::Digest::SHA256`

That's not related. Setting `hash_digest_class` doesn't update the cache client digest, which is `MD5` by default.

```
[4] pry(main)> Rails.cache
=> #<ActiveSupport::Cache::MemCacheStore:0x000000000bd3e428
 @data=
  #<Dalli::Client:0x000000000bd3e1d0
   @key_manager=
    #<Dalli::KeyManager:0x000000000bd3e0b8
     @key_options={:digest_class=>Digest::MD5},
     @namespace=nil>,
   @normalized_servers=["localhost:11211"],
   @options={},
   @ring=nil>,
```

Reply:
@mayorova these are the docs for the Dalli Client, where `:digest_class` is defined:

https://github.com/petergoldstein/dalli/wiki/Dalli::Client-Options

Reply:
Thank you for the info @jlledom !

Well if both you and @akostadinov prefer to have it in the config file, that's fine. 

I just think if that's a "fixed" value that is not going to change, having it as a default value (maybe keep the option for overriding it via the config file) is more practical, so we don't need to carry it over to all the configuration files.

Reply:
I've been trying. The digest can't be overridden from an initializer, but it can be easily overridden in `application.rb`. I don't like it, but it's only one line. Anyway I think we should keep it in the configuration files, because it's configuration. Customers can be using other cache stores for which `:digest_class` means nothing (It's a Dalli exclusive option). I don't think it's a fixed value, it's our config.

Reply:
@akostadinov your opinion on :point_up: ?

I understand that @jlledom suggests just to set `:digest_class: !ruby/class Digest::SHA256` in the `cache_store.yml` config file.

I suggest to make this setting optional in the config file, and set the default value in `config/application.rb` (so, if you don't set anything in the config file `Digest::SHA256` will be used, but you can override it)

Reply:
IIRC MD5 was hardcoded in dalli. And we better update it now we don't support RUby 2.4 anymore. @jlledom are you sure your setting makes any difference for our version of dalli?

Reply:
I have no hard opinion where this configuration should be set. We only really care in FIPS mode where MD5 will fail anyway. This configuration should not be critical elsewhere.

Reply:
In order to unlock this, I pushed https://github.com/3scale/porta/pull/3456/commits/982b1cae28b1875facee10df996951a0801ed295. However, I found that the Rails cache management in this branch seems to be broken, not all cache stores work. I'll check this.